internal/app/machined/pkg/runtime/v1alpha1/platform/packet/packet.go

```diff
@@ -29,10 +92,110 @@ func (p *Packet) Name() string {
 }
 
 // Configuration implements the platform.Platform interface.
+// nolint: gocyclo
 func (p *Packet) Configuration(ctx context.Context) ([]byte, error) {
+	// Fetch and unmarshal both the talos machine config and the
+	// metadata about the instance from packet's metadata server
 	log.Printf("fetching machine config from: %q", PacketUserDataEndpoint)
 
-	return download.Download(ctx, PacketUserDataEndpoint)
+	machineConfig, err := download.Download(ctx, PacketUserDataEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("fetching equinix network config from: %q", PacketMetaDataEndpoint)
+
+	metadataConfig, err := download.Download(ctx, PacketMetaDataEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	var unmarshalledMetadataConfig Metadata
+	if err = json.Unmarshal(metadataConfig, &unmarshalledMetadataConfig); err != nil {
+		return nil, err
+	}
+
+	unmarshalledMachineConfig := &v1alpha1.Config{}
+	if err = yaml.Unmarshal(machineConfig, unmarshalledMachineConfig); err != nil {
```

Comment:
my suggestion here would be to use `configloader.NewFromBytes(machineConfig)` and typecast result to `(*v1alpha1.Config)` to modify it in place

Reply:
oh that's a good idea. will do.

Reply:
k - updated! tested it out on packet and seems to work fine.